tendermint/src/lite_impl/signed_header.rs

```diff
@@ -35,7 +35,13 @@ impl lite::Commit for block::signed_header::SignedHeader {
             let sign_bytes = vote.sign_bytes();
 
             if !val.verify_signature(&sign_bytes, vote.signature()) {
-                return Err(Kind::InvalidSignature.into());
+                return Err(Kind::ImplementationSpecific
+                    .context(format!(
+                        "Couldn't verify signature {:?} with validator {:?}",
```

Comment:
Should we include the sign bytes in here too ?

Reply:
Added in 8c99269bec65fcc4dbe9e9134562fba3a5f0cb27

Reply:
Note that if you do use anomaly, this whole thing can be replaced by:

```rust
use anomaly::fail;
fail!(Kind::ImplementationSpecific, "Couldn't verify signature {:?} with validator {:?} on sign_bytes {:?}", vote.signature(), val, sign_bytes);
```

Reply:
Aha, good to now, will use `fail!` here and everywhere else then. As always, cool library BTW! Maybe the example doc could be slightly more elaborate.